packages/back-end/src/api/organizations/deleteEnvironment.ts

```diff
@@ -0,0 +1,49 @@
+import { DeleteEnvironmentResponse } from "../../../types/openapi";
+import {
+  createApiRequestHandler,
+  validateIsSuperUserRequest,
+} from "../../util/handler";
+import { deleteEnvironmentValidator } from "../../validators/openapi";
+import {
+  findOrganizationById,
+  updateOrganization,
+} from "../../models/OrganizationModel";
+import { OrganizationInterface } from "../../../types/organization";
+
+export const deleteEnvironment = createApiRequestHandler(
+  deleteEnvironmentValidator
+)(
+  async (req): Promise<DeleteEnvironmentResponse> => {
+    await validateIsSuperUserRequest(req);
+
+    const id = req.params.id;
+    const environmentId = req.params.environmentId;
+
+    const org = await findOrganizationById(id);
+    if (!org) {
+      throw Error("Organization not found");
+    }
+
+    const environments = org.settings?.environments || [];
+
+    const environment = environments.find((env) => env.id === environmentId);
+    if (!environment) {
+      throw Error(`Environment ${environmentId} does not exists!`);
+    }
```

Comment:
Missing permission checks.

```ts
if (!req.context.permissions.canDeleteEnvironment(environment)) {
  req.context.permissions.throwPermissionError();
}
```

There is an equivalent `canCreateOrUpdateEnvironment` method for the other endpoints as well